src/routes/referral.js

```diff
@@ -22,10 +21,9 @@ router.post("/community", async function (req, res) {
 });
 
 router.post("/facility", async function (req, res) {
-  setGlobalInstance(req);
-  const { status, serviceRequestId } = await createFacilityReferral(req.body);
+  const { status, serviceRequestId } = await createFacilityReferral(req.body,res);
   res.setHeader("Content-Type", "application/json");
-  res.status(status).send(
+  res.status(status || 500).send(
```

Comment:
The function should return the status code instead of relying on a default

Reply:
There is an issue with the passed statuses being undefined.